bubble_sort.rb

```diff
@@ -0,0 +1,37 @@
+def bubble_sort(arr)  
+  arr.each { |element|
+    arr.each_with_index { |element,index|
+    unless index == arr.length - 1
+    if arr[index] > arr[index + 1]        
+      tmp = arr[index]
+      arr[index] = arr[index + 1]
+      arr[index + 1] = tmp
+    end
+    end
+    }
+  }
+end
+
+def bubble_sort_by(array)
+  for i in 0...array.length
+  for j in 0...array.length - 1 
+    check = yield(array[j],array[j + 1])
+    if check > 0
+    temp = array[j]
```

Comment:
You can put lines 18 and 19 in the same line, without the need for extra variable `if yield(array[j], array[j + 1]) > 0`